_posts/2019/2019-07-24-how-to-transpile-node-modules-with-babel-and-webpack-in-a-monorepo.md

```diff
@@ -0,0 +1,140 @@
+
+### What is a monorepo?
+
+A [monorepo](https://en.wikipedia.org/wiki/Monorepo){:target="_blank"}, short for *monolithic repository*, is  a software development strategy where code for several projects is stored in a single repository.
+The advantages of using a monorepo structure are numerous and include simplified dependency management, streamlining of large scale refactoring processes and ease of code reuse among others.
+Despite all the vaunted advantages, a glaring flaw of the monorepo project structure which affected us in this case is the fact that it complicates build and version control processes.
+There aren't many tools that help overcome the version control issues, and this has led to companies such as Facebook and Microsoft contributing heavily to or forking their own versions of their preferred version control software, or just building theirs as in the case of Google.
+Build problems are in a similar situation, so much so that Facebook and Google also developed their own build software.
+
+We are fortunate to have [Lerna](https://github.com/lerna/lerna){:target="_blank"}, a tool which optimizes the workflow around monorepos with git and npm, and for this reason fits our purposes quite nicely.
+The way Lerna sets up monorepos, most `node_modules` folders end up in the root folder, but are referenced in the `package.json` of their respective packages.
+`are-you-es5` is currently not configured to work with such a project structure.
+
+## Enter depoulo/are-you-es5
+
+[Paolo Priotto](https://github.com/depoulo){:target="_blank"} forked the `are-you-es5` repository and changed it so instead of hardcoding a path to the `node_modules` directory, `require.resolve()` is used to generate an absolute path to each package's directory.
+The pull request is [here](https://github.com/obahareth/are-you-es5/pull/12){:target="_blank"}.
+The performance caveats that ensue due to the fact that `require.resolve()` interacts with the file system are acceptable, since this occurs only at build time.
+Also, it worked!!
+Using this modified version we are able to check if all our dependencies are transpiled or not, but it doesn't end there.
+There is an ongoing discussion on the pull request, but so far our solution has worked reliably for us. 
+
+## Our final solution
+
+We added Paolo's version of `are-you-es5` to `devDependencies` in the package with the untranspiled modules.
+
+```
+yarn add depoulo/are-you-es5#master-but-with-all-of-depoulos-pull-requests-merged -D
+```
+
+Or if you use npm,
+
+```
+npm install depoulo/are-you-es5#master-but-with-all-of-depoulos-pull-requests-merged --save-dev
+```
+
+Next, we created a file called `non_ES5_node_modules` in the root of the package.
+Then we added a `postinstall` script to the scripts object of our package's `package.json` like so
```

Comment:
_Completely optional_
```suggestion
Then we added a `postinstall` script to the scripts object of our package's `package.json` like this:
```

Reply:
IMHO, `like this:` is more appropriate.
If you insist in `so`, please still add the colon `;`